### ISSUE TITLE
How to modify a tls program by progamming before a simulation starts

### ISSUE THREAD

**Issue by nobugforerer** [eclipse-sumo/sumo#14633]
using traci or sth else to change its cycles , phase duation etc.

**Comment by namdre** [comment]
did you already read https://sumo.dlr.de/docs/Simulation/Traffic_Lights.html ?

**Comment by nobugforerer** [comment]
> did you already read https://sumo.dlr.de/docs/Simulation/Traffic_Lights.html ?

yes, it seems that i need to modify it in configuration, but i want a random cycle, a random duation before every simulation starts

**Comment by namdre** [comment]
you can either patch the program via TraCI or generate a new xml definition and load it as additional-file in when calling traci.start (or add the file to your .sumocfg).
